nextcord/state.py

```diff
@@ -1377,11 +1377,13 @@ def parse_interaction_create(self, data) -> None:
         if data["type"] == 3:  # interaction component
             custom_id = interaction.data["custom_id"]  # type: ignore
             component_type = interaction.data["component_type"]  # type: ignore
-            self._view_store.dispatch(component_type, custom_id, interaction)
+            self._view_store.dispatch(component_type, custom_id, interaction)  # type: ignore
+            # Some Interaction types incompatible with this method -> accounted for via if-statement
```

Comment:
Not in Pyright's eyes. Checking if a certain value in the raw interaction data used to make an `Interaction` is another value doesn't mean that the created object is magically that type. You have to add more to the if statement to justify this.